mne/time_frequency/psd.py

```diff
@@ -128,16 +145,21 @@ def psd_array_welch(x, sfreq, fmin=0, fmax=np.inf, n_fft=256, n_overlap=0,
                              for d in x_splits)
 
     # Combining, reducing windows and reshaping to original data shape
-    psds = np.concatenate([np.nanmean(f_s, axis=-1)
-                           for f_s in f_spectrogram], axis=0)
-    psds.shape = np.hstack([dshape, -1])
+    if combine is not None:
```

Comment:
combine is not None by default so you should not need to branch here. Only have _get_reduction return np.mean when 'mean' is passed.